### PR TITLE
Multi instance intro usability fixes

### PR DESCRIPTION
**Pull request by moxiegirl**
## Pre-flight checklist

- [x ] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

This a start to fix a bug I opened: https://github.com/facebook/docusaurus/issues/8827. I would have rewritten the whole page, but your contributor guidelines stipulate "small PRs of less than 500 lines" so...  And, of course, it takes a lot of time to fix a page like this -- and if the issue isn't accepted or this PR languishes ---  well so it goes. Best to start conservatively -- only have so much free time in the world.

## Test Plan

I consider .md code. The `test plan` is to build the site locally. I attempted to `run npm-start` on the `website` dir in my local fork. Seemed obvious but failed miserably --- I'm sure you have a build file instead \_(ツ)_/¯.  I checked your contributor guidelines but there was no mention of building the `facebook/docusaurus /website` locally.  Bummer, as I didn't have time this morning to figure that one out.

### Test links

https://deploy-preview-8828--docusaurus-2.netlify.app/docs/docs-multi-instance/

### CODE REVIEW

**Comment by moxiegirl** [comment]
yeah, sorry for not squashing the commits on this one.

**Comment by slorber** [comment]
Thanks, will review this soon

---

Test link: https://deploy-preview-8828--docusaurus-2.netlify.app/docs/docs-multi-instance/

---

> I checked your contributor guidelines but there was no mention of building the facebook/docusaurus /website

CONTRIBUTING.md contains this:

```md
### Installation

1. Ensure you have [Yarn](https://yarnpkg.com/) installed.
2. After cloning the repository, run `yarn install` in the root of the repository. This will install all dependencies as well as build all local packages.
3. To start a development server, run `yarn workspace website start`.
```

https://docusaurus.io/community/contributing/#installation

--- 

> yeah, sorry for not squashing the commits on this one.

doesn't matter for us, we squash on merge

File: website/docs/guides/docs/docs-multi-instance.mdx [hunk 1]
```diff
@@ -1,50 +1,45 @@
...
 id: multi-instance
-description: Use multiple docs plugin instances on a single Docusaurus site.
+description: Use the multi-instance feature to serve 2 or more doc sets.
 slug: /docs-multi-instance
 ---
 
 # Docs Multi-instance
 
-The `@docusaurus/plugin-content-docs` plugin can support [multi-instance](../../using-plugins.mdx#multi-instance-plugins-and-plugin-ids).
+The `@docusaurus/plugin-content-docs` plugin supports [multi-instance](../../using-plugins.mdx#multi-instance-plugins-and-plugin-ids). The multi-instance feature allows a site to serve 2 or more distinct sets of documentation from the same Docusaurus source.
```
Comment (slorber) [inline website/docs/guides/docs/docs-multi-instance.mdx:9]:
```suggestion
The `@docusaurus/plugin-content-docs` plugin supports [multi-instance](../../using-plugins.mdx#multi-instance-plugins-and-plugin-ids). The multi-instance feature allows a site to serve 2 or more distinct sets of documentation from the same Docusaurus site.
```

I don't know what "Docusaurus source" means, that feels a bit abstract to me.

File: website/docs/guides/docs/docs-multi-instance.mdx [hunk 2]
```diff
@@ -1,50 +1,45 @@
...
 
 :::note
 
-This feature is only useful for [versioned documentation](./versioning.mdx). It is recommended to be familiar with docs versioning before reading this page. If you just want [multiple sidebars](./sidebar/multiple-sidebars.mdx), you can do so within one plugin.
+If you only want multiple sidebars, use a `sidebar` plugin instead. See [Using multiple sidebars](./sidebar/multiple-sidebars.mdx) for how to do this.
```
Comment (slorber) [inline website/docs/guides/docs/docs-multi-instance.mdx:13]: I still believe that it's mostly useful for versioned docs set. If you have iOS/Android SDKs but they are not versioned, you can just use one plugin instance for both and that's enough. `./docs/ios` + `./docs/android` on a single instance (even if the doc sets are large)

> The Note is misleading. I don't need to use versioning to use this feature. It simply makes sense in many use cases.

Please describe the many use cases

---

What is a "sidebar plugin" you refer to here? This concept does not exist. 

The former docs said you could use multiple sidebars within ONE docs plugin (ie no concept of sidebar plugin)

File: website/docs/guides/docs/docs-multi-instance.mdx [hunk 3]
```diff
@@ -1,50 +1,45 @@
...
 
 :::
 
 ## Use-cases {#use-cases}
 
-Sometimes you want a Docusaurus site to host 2 distinct sets of documentation (or more).
+These are 2 of the everyday use cases for the multi-instance feature.
 
-These documentations may even have different versioning/release lifecycles.
```
Comment (slorber) [inline website/docs/guides/docs/docs-multi-instance.mdx:21]: I still think it's mostly useful for versioned docs. Clarifying your use cases with a very concrete example first will be important.

There are certainly cases where 2 unversioned plugins make sense but these are not the most common cases

One use-case I can think of is if you have 2 doc sets for which you need significantly different plugin options.

File: website/docs/guides/docs/docs-multi-instance.mdx [hunk 4]
```diff
@@ -1,50 +1,45 @@
...
-- iOS SDK documentation (`v1.0`, `v2.0`)
-
-In this case, you can use a distinct docs plugin instance per mobile SDK documentation.
-
-:::caution
-
-If each documentation instance is very large, you should rather create 2 distinct Docusaurus sites.
+With large documentation sets, rebuilding may require extra work to verify and stage the set that did not change. By separating the two sets you can build only the set that changed.
 
-If someone edits the iOS documentation, is it really useful to rebuild everything, including the whole Android documentation that did not change?
```
Comment (slorber) [inline website/docs/guides/docs/docs-multi-instance.mdx:36]: The previous advice was to create 2 separate Docusaurus sites, not using 2 separate Docusaurus instances. It's not the same advice anymore.

I don't know what "verify and stage the set that did not change" means. Docusaurus builds full sites, it does not "stage" documents nor can build a site partially

**Review by slorber — CHANGES_REQUESTED** [review]
Thanks

I agree that we can improve this page, but I don't really agree with your opinions and all your changes so far

**Comment by moxiegirl** [comment]
Hi @slorber thank you for the clarification on some key points.   I could certainly revise this for your comments.  Would you like me to do that?  It is a small PR and I can close it if the revision review would take you more bother than it is worth.

**Comment by slorber** [comment]
> Hi @slorber thank you for the clarification on some key points. I could certainly revise this for your comments. Would you like me to do that? It is a small PR and I can close it if the revision review would take you more bother than it is worth.

Reviewing large doc refactors usually takes some time, yes, but I'm open to improving things that we both agree to improve. Maybe it's better to explain your use case first and then we can figure out together what's missing in the doc that would have helped you

**Comment by moxiegirl** [comment]
@slorber  I did explain my use case. See the issue listed in the PR description.

**Comment by slorber** [comment]
> @slorber I did explain my use case. See the issue listed in the PR description.

I just read it again and no you didn't explain your use-case.

You only said that this feature is useful for 2 unversioned docs plugin instances but never justify why.

I'm expecting you to give me a very concrete example here. Give me a sample of the config you are using, give me real plugin id values, give me your business use-case and the outcomes you expect by having 2 instances instead of 1, and why using just 1 was not possible or is not good enough

**Comment by moxiegirl** [comment]
Hi @slorber thank you for your reply and your time. I'll close this PR and the issue.  It isn't that I don't think either is valid it is simply that the barrier to entry is higher than I have time for right now. 

The vast majority of my questions about Docusaurus multi instance I was able to answer with external blogs. I'll write one that consolidates what I learned. Maybe that will help other people who are in my position. Thanks again for your time.

**Comment by slorber** [comment]
I understand @moxiegirl , let's close for now and see later, maybe others will report similar issues and we'll have better insight on how to improve the docs. We can also focus on editing smaller part of the docs that we agree on first.

Still interested to understand better your use-case and why a single unversioned instance is not a good fit

**Comment by moxiegirl** [comment]
Hi @slorber,

I'll write up that blog to illustrate my perspective.  I think a whole page example could give you a better understanding of what I'm trying to convey.  Documentation is often closer to design. Just as when designing a logo, stopping the process at each incremental step to choose a color or general shape can impede the flow. In short, you end up with a crippled camel instead of an elegant horse.  

I'll send you that blog when it is done.

Mary

**Pull request closed without merging**